internal/processor/manifests.go

```diff
@@ -174,7 +174,7 @@ func (p *Processor) ValidateExistingManifest(ctx context.Context, account models
 		return err
 	}
 
-	return p.validateAndStoreManifestCommon(ctx, account, repo, manifest, manifestBytes,
+	return p.validateAndStoreManifestCommon(ctx, account, repo, manifest, manifest.Digest, manifestBytes,
```

Comment:
By passing `manifest.Digest` here, you are neutering the digest check in `validateAndStoreManifestCommon` because it checks `manifest.Digest == manifest.Digest`, which can never fail.

You need to actually compute the digest from `manifestBytes` here. Otherwise, the storage corruption check portion of ValidateManifestsJob is disabled.

I actually recommend removing the `digest` argument entirely from `validateAndStoreManifestCommon`, and having it compute the digest from `manifestBytes` (like `distribution.UnmarshalManifest()` used to do).

Reply:
> I actually recommend removing the `digest` argument entirely from `validateAndStoreManifestCommon`, and having it compute the digest from `manifestBytes` (like `distribution.UnmarshalManifest()` used to do).

Then we calculate it in ValidateAndStoreManifest twice.